kolibri/plugins/learn/assets/src/state/actions.js

```diff
@@ -259,11 +259,12 @@ function showExploreContent(store, channelId, id) {
 }
 
 
-function showLearnChannel(store, channelId, page = 1) {
+function showLearnChannel(store, channelId, cursor) {
   // Special case for when only the page number changes:
   // Don't set the 'page loading' boolean, to prevent flash and loss of keyboard focus.
   const state = store.state;
-  if (state.pageName !== PageNames.LEARN_CHANNEL || state.currentChannel !== channelId) {
+  if (state.pageName !== PageNames.LEARN_CHANNEL ||
+    coreGetters.getCurrentChannelId(state) !== channelId) {
```

Comment:
We weren't checking the current channel properly according to our current state schema. Use the getter to avoid relying on the particular state structure.